### PR TITLE
Update Travis configuration to include Codecov reporting and new build matrix

### PR DESCRIPTION
**Pull request by robbieaverill**
Part of #82

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,22 +1,33 @@
...
-  - DB=MYSQL CORE_RELEASE=3.1
-  global:
-    secure: HJ7sHJww/NX3PPVGjy0ggZyZSLoAdGB4BZJ0UEnIDNhR1fZ1aF4pAEA1K5cvksV9FGCbQ9Z/2JfRk8csDgWIybkBuckZsj+7yjd8u6AEPLKve+GA/Z3iN21jrDZiCC/gugAWJl7OdmR+64bFBjHbvZzYUusjSjpSRO5EqqWPgWA=
-addons:
-  code_climate:
-    repo_token: edaded3f06ffcb76affdc4e747ad6a8d29093594ad2df74405d000da9c9e6899
+  - vendor/bin/phpunit "$COVERAGE" ideannotator/tests
+
+# Share coverage results with codecov.io
+after_success:
```
Comment (Firesphere) [inline .travis.yml]: what about after_failure?

File: .travis.yml [hunk 2]
```diff
@@ -1,22 +1,33 @@
...
-- composer self-update || true
-- phpenv rehash
-- git clone git://github.com/silverstripe-labs/silverstripe-travis-support.git ~/travis-support
-- php ~/travis-support/travis_setup.php --source `pwd` --target ~/builds/ss
-- cd ~/builds/ss
 language: php
+
 php:
-  [5.3,5.4,5.5,5.6,7.0]
+  - 5.3
```
Comment (Firesphere) [inline .travis.yml]: Although it's SS3 supported, should we still support php <=5.4?

Comment (robbieaverill) [inline .travis.yml]: We need to support the same versions that SS do

File: .travis.yml (hunk 1 above)
Comment (robbieaverill) [inline .travis.yml]: We don't want coverage reports for failed builds. BTW this is copied through from https://github.com/silverleague/silverleague.github.io/wiki/Repository-templates and modified for SS3

File: .travis.yml (hunk 2 above)
Comment (Firesphere) [inline .travis.yml]: Okay, reflectiondocblock supports php 5.3.3... Very uncomfortably I'm saying "okay"

Comment (robbieaverill) [inline .travis.yml]: We could enforce that version as a minimum in the composer requirements

File: .travis.yml (hunk 1 above)
Comment (Firesphere) [inline .travis.yml]: Just making sure we're not missing anything. Other than that, if the build passes, I'm good :)

File: .travis.yml (hunk 2 above)
Comment (Firesphere) [inline .travis.yml]: That would make a conflict in Travis, requiring allowed failure. I would prefer suggested in this case (and it still hurts to agree to <5.5 :P )

**Pull request merged**